### PR TITLE
Makes the homepage nodes editable in a UI.

### LINKED ISSUE #1996: Home entityreference (homepage)
Add the existing `field_campaigns` entityreference field to the `home` content type.

Alter the `paraneue_dosomething_preprocess_node` logic to use the field content to determine what to display, instead of a hardcoded array.

I'd argue that this preprocess code should go into a `dosomething_home.theme.inc` vs the theme.

See #1994

### PR DESCRIPTION
**Pull request by angaither**
- adds field_campaign to dosomething_home
- upped the version number for dosomething_home
- moved preprocess for the homepage into the module

Fixes #1996

### CODE REVIEW

File: lib/modules/dosomething/dosomething_home/dosomething_home.module [hunk 1]
```diff
@@ -5,3 +5,45 @@
...
+  $campaigns = $vars['field_campaigns'];
+
+  foreach($campaigns as $key => $campaign) {
+    // Get the campaign nid.
+    $nid = $campaign['target_id'];
+    $node = node_load($nid);
+
+
+    // Grab all campagin attributes.
+    $campagin['nid'] = $nid;
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_home/dosomething_home.module]: Misspelled.  Is this working?

**Comment by barryclark** [comment]
Cooooool, looking forward to seeing this! :dancers:

**Comment by DFurnes** [comment]
OH GOODNESS YES

**Comment by aaronschachter** [comment]
:+1: ship it value added

**Comment by barryclark** [comment]
@angaither let's check this out on staging and maybe we can get it into prod deploy today?

**Pull request merged**